euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/MultiDataSink.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright 2016-2017 Seznam.cz, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.seznam.euphoria.core.client.io;
+
+import cz.seznam.euphoria.core.client.functional.UnaryFunction;
+import cz.seznam.euphoria.core.util.IOUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ *
+ * MultiDataSink allows to save to multiple {@link DataSink}
+ *
+ * Example usage:
+ *<pre>
+ * {@code
+ * interface InputElement {
+ *   enum Type { FIRST, SECOND, THIRD }
+ *   Type getType();
+ *   DataFirst getFirst();
+ *   DataSecond getSecond();
+ *   DataThird getThird();
+ * }
+ *
+ * DataSink<InputElement> = MultiDataSink
+ *   .selectBy(InputElement::getType)
+ *   .addSink(Type.FIRST, InputElement::getFirst, new SequenceFileSink<>(...))
+ *   .addSink(Type.SECOND, InputElement::getSecond, new StdoutSink<>(...))
+ *   .addSink(Type.THIRD, InputElement::getThird, new StdoutSink<>(...))
+ *   .build();
+ * }
+ *</pre>
+ * @param <SPLIT> key to select DataSink where saves the element
+ * @param <IN> type of input element
+ */
+public class MultiDataSink<SPLIT, IN> implements DataSink<IN> {
+
+  private final UnaryFunction<IN, SPLIT> splitFunction;
+  private final Map<SPLIT, DataSinkWrapper<SPLIT, IN, ?>> sinks;
+
+  private MultiDataSink(
+      UnaryFunction<IN, SPLIT> splitFunction,
+      Map<SPLIT, DataSinkWrapper<SPLIT, IN, ?>> sinks) {
+    this.splitFunction = splitFunction;
+    this.sinks = sinks;
+  }
+
+  /**
+   * Selects DataSink where to save output elements
+   * @param splitFunction transform
+   * @param <SPLIT> key to select DataSink
+   * @param <IN> type of input element
+   * @return Builder
+   */
+  public static <SPLIT, IN> Builder<SPLIT, IN> selectBy(UnaryFunction<IN, SPLIT> splitFunction) {
```

Comment:
how about renaming splitFunction to selectFunction?

Reply:
:+1: